magellan-library/src/main/java/com/wealthfront/magellan/navigation/NavigationDelegate.kt

```diff
@@ -35,7 +36,7 @@ public class NavigationDelegate(
       updateMenu(menu)
     }
 
-  public val backStack: Stack<NavigationEvent> = Stack()
+  public val backStack: Deque<NavigationEvent> = ArrayDeque()
```

Comment:
Changed from Stack to ArrayDeque for backwards compatibility, there should be virtually no API difference..

Reply:
I forget how this is structured, but is there a way we can contain this change to the `magellan-legacy` artifact? I think `Stack` is a better model for what we're trying to do here and would like to use that moving forward if possible. I don't feel that strongly though.

Reply:
Hmm. It would be weird to have different data structures for the same functionality between magellan-library and magellan-legacy. I was thinking about the pros and cons and there doesn't seem to be anything that swings me either way so I thought i can keep the migration and thinking simpler to keep the existing `ArrayDeque` that we were using in 1.0.